seismostats/analysis/b_significant.py

```diff
@@ -0,0 +1,172 @@
+"""This module contains functions for the evaluation if a one dimensional
+b-value series varies significantly, following Mirwald et al, SRL (2024)
+"""
+
+import numpy as np
+import datetime as dt
+import warnings
+
+from seismostats.analysis.bvalue import ClassicBValueEstimator
+from seismostats.analysis.bvalue.base import BValueEstimator
+
+
+def est_morans_i(values: np.ndarray, w: np.ndarray | None):
+    """
+    Estimate the nearest neighbor auto correlation (Moran's I) of the values.
+
+    Args:
+        values:     values
+        w:          Weight matrix, indicating which of the values are
+                neighbors to each other. It should be a square matrix of
+                size len(b_vec) x len(b_vec). At places where the value is nan,
+                the matrix is set to zero, effectively not counting these data
+                points. If w is None, it is assumed that the series of values is
+                1-dimensional, and the values are sorted along that dimension.
+
+    Returns:
+        ac:     Auto correlation of the values
+        n_p:    Sum of the weight matrix. In the limit of a large n (number of
+            values), the upper limit of the standard deviation of the
+            autocorrelation is 1/sqrt(n_p).
+
+    """
+
+    ac = 0
+    ac_0 = 0
+
+    # in case w is not provided, the 1-dimensional case is assumed
+    if w is None:
+        n_values = len(values)
+        w = np.zeros(n_values, n_values)
+        for ii in range(n_values):
+            for jj in range(n_values):
+                if ii == jj + 1:
+                    w[ii, jj] = 1
+
+    # estimate mean
+    n = len(values[~np.isnan(values)])
+    mean_v = np.mean(values[~np.isnan(values)])
+
+    for ii, v1 in enumerate(values):
+        if np.isnan(v1):
+            w[ii, :] = 0
+            continue
+        ac_0 += (v1 - mean_v) ** 2
+        for jj, v2 in enumerate(values):
+            if np.isnan(v2):
+                w[ii, jj] = 0
+                continue
+            if w[ii, jj] == 1:
+                ac += (v1 - mean_v) * (v2 - mean_v)
+
+    n_p = np.sum(w)
+    ac = (n - 1) / n_p * ac / ac_0
+    return ac, n, n_p
+
+
+def transform_n(
+    x: np.ndarray, b: float, n1: np.ndarray, n2: np.ndarray
+) -> np.ndarray:
+    """transform b-value to be comparable to other b-values
+
+    Args:
+        x:  b-value estimates to be transformed
+        b:  true b-value
+        n1:   number of events used for the the b-value estimates
+        n2:   number of events to which the distribution is transformed
+
+    Returns:
+        x:  transformed b-values
+    """
+    x_transformed = b / (1 - np.sqrt(n1 / n2) * (1 - b / x))
+    return x_transformed
+
+
+def b_samples(
```

Comment:
this is basically another b-value estimator, no? Similar to Kijko or Weichert? I will add an implementation for those two methods later on, is it okay if I treat this one the same way?

Reply:
No, this one just uses the b-value estimator in order to create a series of b-values. maybe I should think of a better name..

Reply:
but the description says "Estimate the b-values for a given partition of the data", it literally says "estimate the b-values" .... :D

Reply:
Yes, but it says b-values, that is the plural. The input is also not an array of magnitudes, but a list of arrays.. It's supposed to be an internal function

Reply:
renamed it and tried to make it more clear in the documentation :)